src/openvic-simulation/military/UnitInstance.hpp

```diff
@@ -0,0 +1,101 @@
+#pragma once
+
+#include "openvic-simulation/map/Province.hpp"
+#include "openvic-simulation/military/Leader.hpp"
+#include "openvic-simulation/military/UnitType.hpp"
+#include "openvic-simulation/types/fixed_point/FixedPoint.hpp"
+#include <string_view>
+#include <type_traits>
+#include <vector>
+
+namespace OpenVic {
+	template<typename T, typename std::enable_if<std::is_base_of<UnitType, T>::value>::type* = nullptr>
```

Comment:
```suggestion
	template<std::derived_from<UnitType> T>
```
You can use concepts to avoid needing an extra parameter